internal/handler/api.go

```diff
@@ -34,15 +36,15 @@ func NewAPI(cfg config.Config, database *gorm.DB) *gin.Engine {
 			}
 			users := v1.Group("/users")
 			{
-				admin := users.Use(middleware.JWTAuth(cfg.JWTTokenSecret), middleware.AccessCheck(adminRole))
+				admin := users.Use(middleware.JWTAuth(cfg.JWTTokenSecret), middleware.AccessCheck("admin"), middleware.Cache(cache, time.Duration(cfg.Redis.ShortTtl)*time.Minute))
```

Comment:
NIT: You should to specify TTL time unit either in documentation or/and (even better) explicitly naming your variable. e.g. `CACHE_DURATION_MINUTES` (or something like that) for easier understanding of purpose for both end users and other developers. If it's not explicitly stated, the other developer might use seconds on another endpoint instead of minutes you intended, which will lead to some strange bugs.